internal/versions/rest_api_versions.go

```diff
@@ -48,17 +48,16 @@ func (h *restAPIVersionsHandler) GetReleaseImage(_ context.Context, openshiftVer
 	if cpuArchitecture == common.MultiCPUArchitecture {
 		query = query.Where("cardinality(cpu_architectures) > 1")
 	} else {
-		query = query.Where(h.db.Session(&gorm.Session{}).Where("? = ANY(cpu_architectures)", cpuArchitecture).Or("cpu_architecture = ?", cpuArchitecture))
+		query = query.Where(h.db.Session(&gorm.Session{}).
+			Where("? = ANY(cpu_architectures) AND cardinality(cpu_architectures) = 1", cpuArchitecture).
```

Comment:
Is there some operator we can use here other than both of these?
Can we compare against a literal array?

Reply:
Yes we can, I changed it